### PR TITLE
Support marker="none" to mean "no marker".

### PR DESCRIPTION
**Pull request by anntzer**
As for other APIs (#19300), this is less likely to be confused with the None
object.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: doc/users/next_whats_new/marker_none.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+``marker`` can now be set to the string "none"
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+... to mean "no-marker", consistently with other APIs which support the
+lowercase version.
```
Comment (timhoffm) [inline doc/users/next_whats_new/marker_none.rst:4]:
```suggestion
... to mean *no-marker*, consistently with other APIs which support the
lowercase version. Using 'none' is recommended over using 'None'.
```

**Review by timhoffm — APPROVED** [review]

File: doc/users/next_whats_new/marker_none.rst (hunk 1 above)
Comment (anntzer) [reply doc/users/next_whats_new/marker_none.rst:4]: done

File: lib/matplotlib/markers.py [hunk 2]
```diff
@@ -45,7 +45,8 @@
 ``9`` (``CARETRIGHTBASE``)     |m34|  caretright (centered at base)
 ``10`` (``CARETUPBASE``)       |m35|  caretup (centered at base)
 ``11`` (``CARETDOWNBASE``)     |m36|  caretdown (centered at base)
-``"None"``, ``" "`` or  ``""``        nothing
+``"None"`` or ``"None"``              nothing
```
Comment (jklymak) [inline lib/matplotlib/markers.py:48]: Did you mean:
```suggestion
``"none"`` or ``"None"``              nothing
```

Comment (anntzer) [reply lib/matplotlib/markers.py]: oops, yes.

**Review by jklymak — APPROVED** [review]

**Pull request merged**